### PR TITLE
 JVM Image Fixes

### PR DESCRIPTION
**Pull request by Ershany-IBM**
- Fixed the location where we want to mmap
- Updated the JVMImageHeader
- Fixed the read function in JVMImageHeader

### CODE REVIEW

**Review by VinoshanT — APPROVED** [review]

File: runtime/vm/JVMImage.cpp [hunk 1]
```diff
@@ -323,23 +322,32 @@ JVMImage::readImageFromFile(void)
 	OMRPortLibrary *portLibrary = IMAGE_OMRPORT_FROM_JAVAVM(_vm);
 	OMRPORT_ACCESS_FROM_OMRPORT(portLibrary);
 
+	PORT_ACCESS_FROM_JAVAVM(_vm);
```
Comment (akshayben) [inline runtime/vm/JVMImage.cpp]: put this back to our port if this wasn't the cause of the bug

File: runtime/vm/JVMImage.cpp [hunk 2]
```diff
@@ -323,23 +322,32 @@ JVMImage::readImageFromFile(void)
...
 	if (-1 == fileDescriptor) {
 		return false;
 	}
 
 	/* Read image header then mmap the rest of the image (heap) into memory */
-	omrfile_read(fileDescriptor, (void *)_jvmImageHeader, sizeof(JVMImageHeader));
+	/* TODO: Should only read imageAddress and size because that is the only data we need for mmap */
+	JVMImageHeader *imageHeaderBuffer = (JVMImageHeader *)j9mem_allocate_memory(sizeof(JVMImageHeader), J9MEM_CATEGORY_CLASSES); /* TODO: change category */
+	if (NULL == imageHeaderBuffer) {
+		return NULL;
```
Comment (akshayben) [inline runtime/vm/JVMImage.cpp]: return false

Comment (akshayben) [reply runtime/vm/JVMImage.cpp]: or change these to error codes

File: runtime/vm/JVMImage.cpp (hunk 1 above)
Comment (Ershany-IBM) [reply runtime/vm/JVMImage.cpp]: Now that I think about it. I do not see a reason to allocate the JVMImageHeader buffer on the heap. So I will remove this and move the temporary buffer to the stack

**Pull request merged**